OpenSkyPlusApi/DeviceControls.cs

```diff
@@ -1,230 +1,380 @@
+        _logger.LogInfo($"IsHandednessLeft: {isLeftHandedness}");
+    }
+
+    private static void DumpType(string member, Type type)
+    {
+        var sb = new StringBuilder($"Dumping type of {member}: {type.FullDescription()}:" + Environment.NewLine);
+        sb.AppendLine("Interfaces:");
+        foreach (var t in type.GetInterfaces())
+        {
+            sb.AppendLine($"\t{t}");
+            sb.AppendLine("\tProperties:");
+            foreach (var property in t.GetProperties(BindingFlags.Public | BindingFlags.NonPublic |
+                                                     BindingFlags.Instance | BindingFlags.Static))
+                sb.AppendLine(
+                    $"\t\t{property.PropertyType,30} {property.Name,30} - Info: {(property.CanRead ? "Read" : "")} {(property.CanWrite ? "Write" : "")} {property.Attributes}");
+            sb.AppendLine("\tFields:");
+            foreach (var field in t.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance |
+                                              BindingFlags.Static))
+                sb.AppendLine(
+                    $"\t\t{field.FieldType,30} {field.Name,30} - Info: {(field.IsPrivate ? "Private" : "")} {(field.IsPublic ? "Public" : "")}");
+        }
+
+        sb.AppendLine("Properties:");
+        foreach (var property in type.GetProperties(BindingFlags.Public | BindingFlags.NonPublic |
+                                                    BindingFlags.Instance | BindingFlags.Static))
+            sb.AppendLine(
+                $"\t{property.PropertyType,30} {property.Name,30} - Info: {(property.CanRead ? "Read" : "")} {(property.CanWrite ? "Write" : "")}");
+        sb.AppendLine("Methods:");
+        foreach (var method in type.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance |
+                                               BindingFlags.Static))
+        {
+            var parameters = method.GetParameters();
+            var parameterDescriptions = string.Join
+            (", ", method.GetParameters()
+                .Select(x => x.ParameterType + " " + x.Name)
+                .ToArray());
+
+            sb.AppendLine(
+                $"\t{method.ReturnType,30} {method.Name,30}({parameterDescriptions,50}) - Info: {(method.IsPrivate ? "Private" : "Public"),10} {(method.IsStatic ? "Static" : "Instance"),10}");
+        }
+
+        sb.AppendLine("Fields:");
+        foreach (var field in type.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance |
+                                             BindingFlags.Static))
+            sb.AppendLine(
+                $"\t{field.FieldType,30} {field.Name,30} - Info: {(field.IsPrivate ? "Private" : "")} {(field.IsPublic ? "Public" : "")}");
+        _logger.LogInfo(sb);
+    }
+
+    public static event NotificationMonitorStatusChange MessageMonitorReady;
+
+    public static bool ArmMonitor()
+    {
+        _logger.LogInfo("ArmMonitor");
```

Comment:
If these are going to be user-readable (as they should be), I'd make the messages a bit more user friendly. "Monitor armed and ready for shot" or something along those lines so they understand what they are looking at.

Reply:
I would just put this in debug then, might not be useful for the end user. What do you think?